var/spack/repos/builtin/packages/mesquite/package.py

```diff
@@ -28,10 +28,13 @@ class Mesquite(AutotoolsPackage):
 
     def configure_args(self):
         args = [
-            'CC=%s' % self.spec['mpi'].mpicc,
-            'CXX=%s' % self.spec['mpi'].mpicxx,
-            '--with-mpi=%s' % self.spec['mpi'].prefix,
             '--enable-release',
             '--enable-shared',
         ]
+
+        if '+mpi' in self.spec:
+            args.append('CC=%s' % self.spec['mpi'].mpicc)
+            args.append('CXX=%s' % self.spec['mpi'].mpicxx)
+            args.append('--with-mpi=%s' % self.spec['mpi'].prefix)
```

Comment:
Can you add:
```python
else:
    args.append('--without-mpi')
```
Autotools likes to set the defaults based on whether or not dependencies are found on the system, leading to non-deterministic builds.

Reply:
I updated the branch, thanks for the heads-up